ext/http/00_serve.ts

```diff
@@ -595,6 +595,48 @@ function serve(arg1, arg2) {
     options = { __proto__: null };
   }
 
+  const canOverrideOptions = !ObjectHasOwn(options, "path")
+    && !ObjectHasOwn(options, "hostname")
+    && !ObjectHasOwn(options, "port");
+  const env = Deno.permissions.querySync({ name: "env", variable: "DENO_SERVE_ADDRESS" }).state === "granted"
+    && Deno.env.get("DENO_SERVE_ADDRESS");
```

Comment:
This looks very funky. We are already forwarding host and port in https://github.com/denoland/deno/blob/cd4a0072dc3a9c8783e0a864af9bd428552c2407/runtime/worker_bootstrap.rs#L94-L96, I suggest that instead of doing all of this parsing in JS, we do that in Rust and only for `deno serve` subcommand. Otherwise this code will be applied to every single `Deno.serve()` call.